File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& h) {
        
        int left =0;
        int right = h.size()-1;

        int maxA =0;

        while(left <right){
            int curA = min(h[left], h[right]) * (right - left);
            maxA = max(maxA, curA);

            if(h[left] < h[right]){
                left++;
            }else{
                right--;
            }
        }

        return maxA;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] h) {
        int e = h.length -1;

        int s = 0;

        int Maxwater =0;
        while(s < e){
            int curr = Math.min(h[s] , h[e]) * (e-s);
            Maxwater = Math.max(curr, Maxwater);

            if(h[s] < h[e]){
                s++;
            }else{
                e--;
            }
        }

        return Maxwater;
    }
}